tests/mdev_configuration_allocation_test.go

```diff
@@ -354,6 +360,174 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", Serial, decorators.VGPU
 			Eventually(checkAllMDEVCreated(newDesiredMdevTypeName, newExpectedInstancesNum), 3*time.Minute, 15*time.Second).Should(BeInPhase(k8sv1.PodSucceeded))
 
 		})
+		It("should successfully create VMI when guest memory is less than or equal to request memory", func() {
+			By("Creating a Fedora VMI")
+			networkData := cloudinit.CreateDefaultCloudInitNetworkData()
+			vGPU := v1.GPU{
+				Name:       "gpu1",
+				DeviceName: deviceName,
+			}
+
+			vmi = libvmi.New(
+				libvmi.WithRng(),
+				libvmi.WithContainerDisk("disk0", cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithCloudInitNoCloudNetworkData(networkData),
+				libvmi.WithGuestMemory("512Mi"),
+				withGPUDevice(vGPU),
+			)
+
+			createdVmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
+			Expect(err).ToNot(HaveOccurred())
+			vmi = createdVmi
+			libwait.WaitUntilVMIReady(vmi, console.LoginToFedora)
+		})
+		It("should ignore memory-overcommit if the resulting request memory is less than guest memory", func() {
+			By("Applying cluster memory-overcommit")
+			config.DeveloperConfiguration.MemoryOvercommit = 200
+			tests.UpdateKubeVirtConfigValueAndWait(config)
+
+			By("Creating a Fedora VMI")
+			networkData := cloudinit.CreateDefaultCloudInitNetworkData()
+			vGPU := v1.GPU{
+				Name:       "gpu1",
+				DeviceName: deviceName,
+			}
+
+			vmi = libvmi.New(
+				libvmi.WithRng(),
+				libvmi.WithContainerDisk("disk0", cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithCloudInitNoCloudNetworkData(networkData),
+				libvmi.WithGuestMemory("512Mi"),
+				withGPUDevice(vGPU),
+			)
+
+			createdVmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
+			Expect(err).ToNot(HaveOccurred())
+			vmi = createdVmi
+			libwait.WaitUntilVMIReady(vmi, console.LoginToFedora)
+			Expect(vmi.Spec.Domain.Resources.Requests.Memory().String()).To(Equal("4096M"))
```

Comment:
Don't we already have a unit test for that? This e2e test doesn't seem necessary.
Also, how do we end up with 4096M and not 4096Mi?

Reply:
```
Expected
    <string>: 512Mi
to equal
    <string>: 4096M
tests/mdev_configuration_allocation_test.go:412
```